libjulia/backends/evm/EVMCodeTransform.cpp

```diff
@@ -34,6 +36,80 @@ using namespace dev::solidity;
 
 using Scope = dev::solidity::assembly::Scope;
 
+CodeTransform::CodeTransform(
+	julia::AbstractAssembly& _assembly,
+	solidity::assembly::AsmAnalysisInfo& _analysisInfo,
+	Block const& _block,
+	solidity::AssemblyStack::Language _language,
+	bool _allowStackOpt,
+	bool _yul,
+	bool _evm15,
+	ExternalIdentifierAccess const& _identifierAccess,
+	bool _useNamedLabelsForFunctions,
+	int _stackAdjustment,
+	std::shared_ptr<Context> _context
+):
+	m_assembly(_assembly),
+	m_info(_analysisInfo),
+	m_language(_language),
+	m_allowStackOpt((_language == AssemblyStack::Language::StrictAssembly || _language == AssemblyStack::Language::Yul) && _allowStackOpt),
+	m_yul(_yul),
+	m_evm15(_evm15),
+	m_useNamedLabelsForFunctions(_useNamedLabelsForFunctions),
+	m_identifierAccess(_identifierAccess),
+	m_stackAdjustment(_stackAdjustment),
+	m_context(_context)
+{
+	if (m_allowStackOpt)
+		m_references = ReferencesCounter::countReferences(_block);
+}
+
+void CodeTransform::freeVariables()
+{
+	if (!m_allowStackOpt)
+		return;
+
+	for (pair<string const, size_t>& item : m_references)
+	{
+		string const& name = item.first;
+		if (m_scope->identifiers.count(name) && m_references[name] == 0)
+		{
+			Scope::Identifier* id = m_scope->lookup(name);
+			assertThrow(id, CodegenException, "");
+			Scope::Variable const& var = boost::get<Scope::Variable>(*id);
+			solAssert(m_context->variableStackHeights.count(&var) > 0, "");
+			m_unusedStackSlots.insert(m_context->variableStackHeights[&var]);
+			m_context->variableStackHeights.erase(&var);
+			m_references.erase(name);
+		}
+	}
+}
+
+void CodeTransform::increaseReference(string const& _name)
+{
+	if (!m_allowStackOpt)
+		return;
+
+	if (m_references.count(_name))
+	{
+		m_references[_name]++;
+	}
+	else
+	{
+		m_references[_name] = 1;
+	}
```

Comment:
Style: we usually don't use curly brackets around single statements.

Reply:
will update soon